src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.AnonymousTypeRemover.cs

```diff
@@ -33,6 +33,14 @@ public override ITypeSymbol VisitArrayType(IArrayTypeSymbol symbol)
                 return _compilation.CreateArrayTypeSymbol(elementType, symbol.Rank);
             }
 
+            public override ITypeSymbol VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
+            {
+                // TODO(https://github.com/dotnet/roslyn/issues/43890): function pointers could theoretically
+                // have a parameter of an anonymous type if you have a generic function that returns function
+                // pointers, and that was called with an anonymous type.
+                return symbol;
```

Comment:
why don't we just add the support for this now?  it shouldn't be hard.

Reply:
@333fred is already looking at that.

Reply:
Public API stuff (including CreateFunctionPointerTypeSymbol) will be in my next PR.